drivers/storage/ebs/storage/ec2_storage.go

```diff
@@ -0,0 +1,1303 @@
+	if err != nil {
+		return goof.WithError("Error creating tags", err)
+	}
+	return nil
+}
+
+// Wait for volume action to complete (creation, attachment, detachment)
+func (d *driver) waitVolumeComplete(
+	ctx types.Context, volumeID, action string) error {
+	// no volume id inputted
+	if volumeID == "" {
+		return goof.New("Missing volume ID")
+	}
+
+UpdateLoop:
+	for {
+		// update volume
+		volumes, err := d.getVolume(ctx, volumeID, "")
+		if err != nil {
+			return goof.WithError("Error getting volume", err)
+		}
+
+		// check retrieved volume
+		switch action {
+		case waitVolumeCreate:
+			if *volumes[0].State == awsec2.VolumeStateAvailable {
+				break UpdateLoop
+			}
+		case waitVolumeDetach:
+			if len(volumes[0].Attachments) == 0 {
+				break UpdateLoop
+			}
+		case waitVolumeAttach:
+			if len(volumes[0].Attachments) == 1 &&
+				*volumes[0].Attachments[0].State == awsec2.VolumeAttachmentStateAttached {
+				break UpdateLoop
+			}
+		}
+		time.Sleep(1 * time.Second)
+	}
+
+	return nil
+}
+
+// Wait for snapshot action to complete
+// TODO Snapshots are not implemented yet
+/*
+func (d *driver) waitSnapshotComplete(
+	ctx types.Context, snapshotID string) error {
+	if snapshotID == "" {
+		return goof.New("Missing snapshot ID")
+	}
+
+	for {
+		// update snapshot
+		snapshots, err := d.getSnapshot(ctx, "", snapshotID, "")
+		if err != nil {
+			return goof.WithError(
+				"Error getting snapshot", err)
+		}
+
+		// check retrieved snapshot
+		if len(snapshots) == 0 {
+			return goof.New("No snapshots found")
+		}
+		snapshot := snapshots[0]
+		if *snapshot.State == awsec2.SnapshotStateCompleted {
+			break
+		}
+		if *snapshot.State == awsec2.SnapshotStateError {
+			return goof.Newf("Snapshot state error: %s", *snapshot.StateMessage)
+		}
+		time.Sleep(1 * time.Second)
+	}
+
+	return nil
+}
+*/
+
+// Retrieve volume or snapshot name
+func (d *driver) getName(tags []*awsec2.Tag) string {
+	for _, tag := range tags {
+		if *tag.Key == "Name" {
+			return *tag.Value
+		}
+	}
+	return ""
+}
+
+// Retrieve current instance using EC2 API call
+func (d *driver) getInstance() (awsec2.Instance, error) {
+	diInput := &awsec2.DescribeInstancesInput{
+		InstanceIds: []*string{&d.instanceDocument.InstanceID},
+	}
+	resp, err := d.ec2Instance.DescribeInstances(diInput)
+	if err != nil {
+		return awsec2.Instance{}, goof.WithError(
+			"error retrieving instance with EC2 API call", err)
+	}
+
+	return *resp.Reservations[0].Instances[0], nil
+}
+
+// Get volume or snapshot name without config tag
+func (d *driver) getPrintableName(name string) string {
+	return strings.TrimPrefix(name, d.tag()+ebs.TagDelimiter)
+}
+
+// Prefix volume or snapshot name with config tag
+func (d *driver) getFullName(name string) string {
+	if d.tag() != "" {
+		return d.tag() + ebs.TagDelimiter + name
+	}
+	return name
+}
+
+// Retrieve config arguments
+func (d *driver) accessKey() string {
+	if d.config.GetString("ebs.accessKey") != "" {
```

Comment:
Hi @proudh,

Wherever you're following this convention:

```go
if d.config.GetVALUE("key") != DEFAULT {
    return d.config.GetVALUE("key")
}
```

It should be changed to the following to make it more efficient:

```go
if val := d.config.GetVALUE("key")l val != DEFAULT {
    return val
}
```